### PR TITLE
Remapping

### PR DESCRIPTION
**Pull request by MaxUlysse**
- Add new remapping step (possibility to remap BAMs)
- `fastqFiles` renamed to `inputFiles`
- `MapReads` for `remapping` will now convert BAM to FASTQ and feed it to BWA on the fly

## PR checklist
 - [x] PR is made against `dev` branch
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [x] Ensure the test suite passes (`./scripts/test.sh -p docker -t ALL`).
 - [ ] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated

### CODE REVIEW

**Comment by MaxUlysse** [comment]
I think we could improve that by checking if input is bam or fastq, and then do the correct thing

File: main.nf [hunk 1]
```diff
@@ -155,24 +161,39 @@ process MapReads {
...
+  else if (step == 'remapping')
+    """
+    gatk --java-options -Xmx${task.memory.toGiga()}g \
+    SamToFastq \
+    --INPUT=${inputFile1} \
+    --FASTQ=/dev/stdout \
+    --INTERLEAVE=true \
+    --NON_PF=true \
+    | \
+    bwa mem -K 1000000 -p -R \"${readGroup}\" ${extra} -t ${task.cpus} -M ${genomeFile} \
```
[the remark this answers is not in the document]
Comment (MaxUlysse) [reply main.nf]: Oh yes, I totally forgot about that.
I'll do that right away...

**Comment by MaxUlysse** [comment]
The only thing that is really missing is tests for this new feature...
But as this is an edge case, I'm wondering if we should test it or not.
What do you think?

File: main.nf [hunk 2]
```diff
@@ -133,16 +133,17 @@ process RunFastQC {
...
-    set idPatient, status, idSample, idRun, file(fastqFile1), file(fastqFile2) from fastqFilesforFastQC
+    set idPatient, status, idSample, idRun, file(inputFile1), file(inputFile2) from inputFilesforFastQC
 
   output:
     file "*_fastqc.{zip,html}" into fastQCreport
 
   when: step == 'mapping' && !params.noReports
 
   script:
+  inputFiles = SarekUtils.hasExtension(inputFile1,"fastq.gz") ? "${inputFile1} ${inputFile2}" : "${inputFile1}"
```
Comment (alneberg) [inline main.nf]: We should probably allow fq.gz as well right?

Comment (MaxUlysse) [reply main.nf]: We can think of it in the extractSample() function, but currently I can't think of anyone asking me about this particular `.fq.gz` extension.

**Comment by MaxUlysse** [comment]
The index is not in the TSV when mapping BAMs.
I figured that if you want to (re)map BAMs, there's little use of the indexes.
